csharp/src/Ice/Internal/IdleTimeoutTransceiverDecorator.cs

```diff
@@ -126,7 +117,10 @@ internal IdleTimeoutTransceiverDecorator(Transceiver decoratee, ConnectionI conn
 
     internal void enableIdleCheck()
     {
-        if (!idleCheckEnabled && _readTimer is not null)
+        // idleCheckEnabled is already true when enableIdleCheck() is called on a TCP server connection that becomes
```

Comment:
Bug 2:
we can't count on a connection to read anything to schedule the very first idle check.

For example, a TCP server connection can become active and not read anything ever. This should trigger an idle timeout.

Reply:
I don't understand this second fix.

Reply:
Take a TCP server connection, where the client doesn't send any request or ValidateConnection frame. Maybe because it's not an Ice client.

In this case, when do we schedule the very fist idle check? Can't be in a read since we don't read anything. A good spot is when the ConnectionI becomes active for the first time and calls enableIdleCheck.

Reply:
It's set to true by the IdleTimeoutTransceiverDecorator constructor.

Maybe a better fix would be to set it to false in the constructor, and rely on the first enableIdleCheck() (from the first activate) to flip it to true?

Reply:
Fixed as suggested: the constructor no longer sets idleCheckEnabled to true.